src/Nethermind/Nethermind.Evm/EvmStack.cs

```diff
@@ -94,9 +94,12 @@ public void PushByte(byte value)
         _tracer?.ReportStackPush(value);
 
         ref byte bytes = ref PushBytesRef();
-        // Not full entry, clear first
-        Unsafe.As<byte, Word>(ref bytes) = default;
-        Unsafe.Add(ref bytes, WordSize - sizeof(byte)) = value;
+
+        // Build a 256-bit vector: [ 0, 0, 0, (value << 56) ]
+        // - when viewed as bytes: all zeros except byte[31] == value
+
+        // Single 32-byte store: last byte as value
+        Unsafe.As<byte, Word>(ref bytes) = Vector256.Create(0UL, 0UL, 0UL, (ulong)value << 56).AsByte();
```

Comment:
Can you reuse code? There are 4 times this code is reused, 6 times if you count no shift.